lib/engine/game/g_1822/step/special_track.rb

```diff
@@ -99,7 +99,7 @@ def available_hex(entity, hex)
 
             # If player have choosen the tile lay option on the Edinburgh and Glasgow Railway company,
             # only rough terrain, hill or mountains are valid hexes
-            if entity.id == @game.class::COMPANY_EGR
+            if @game.must_be_on_terrain?(entity)
               tile_terrain = hex.tile.upgrades.any? do |upgrade|
                 %i[mountain hill swamp].any? { |t| upgrade.terrains.include?(t) }
```

Comment:
Swamp seems like a bug? According to the P8 description, it is only usable on hills and mountains.

Reply:
This is a bug - but a bigger bug because "hill" in 18xx.games is used for both "hill" and "rough".  I'll write a bug for this and get to that this week.